### PR TITLE
Disable save and delete buttons in QCBX overlay in app resources

### LINKED ISSUE #4886: Can create new records using a query combo box in the visual editor for form definitions
**Describe the bug**
In the visual editior for any form definition if you click the add button on a query combo box you can add a new record which can save to the database. 

**To Reproduce**

1. Go to app resources -> form definition 
2. Click on any table with a query combo box in the form (e.g. collection object with the accession qcbx) 
3. In visual editor click the add button for the query combo box
4. Fill out some information in the form 
5. See you can save 

**Expected behavior**
You should not be able to create a record in the form editor and the button should probably be disabled. 

**Screenshots**

https://github.com/specify/specify7/assets/150940840/d56b339b-4932-4d0c-811b-3bd2227b4495

Please, also fill out the following information manually:

- OS: Windows 11
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: lsumzmammals9823
- Collection name: mammals
- User Name: spadmin
[Specify 7 System Information - 2024-05-03T19_18_29.501Z.txt](https://github.com/specify/specify7/files/15204190/Specify.7.System.Information.-.2024-05-03T19_18_29.501Z.txt)

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4886

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions

- Go to app resources -> form definition
- Click on any table with a query combo box in the form (e.g. collection object with the accession qcbx)
- In visual editor click the add button for the query combo box
- Fill out some information in the form
- [ ] verify save button is disabled 
- [ ] verify there is no delete button

### CODE REVIEW

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]
take a look at how FormMeta displays different dialog based on whether it's in the form editor or not, and do something similar here, rather than URL parsing

**Comment by CarolineDenis** [comment]
@maxpatiiuk , 
Not sure what you mean, it is just not returned from useResourceView(). 
How would this apply to the subforms in app resources?

File: specifyweb/frontend/js_src/lib/components/Forms/Save.tsx [hunk 1]
```diff
@@ -107,8 +108,14 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
   const canUpdate = hasTablePermission(resource.specifyTable.name, 'update');
   const canSave = resource.isNew() ? canCreate : canUpdate;
 
+  const location = useLocation();
+  const currentUrl = location.pathname;
+  const isInViewSet = currentUrl.includes('view-set');
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/Forms/Save.tsx:113]:
```suggestion
  const isInFormEditor = React.useContext(InFormEditorContext);
```

see https://github.com/specify/specify7/blob/71ad206747904a98fac6043116e2c05331f5036c/specifyweb/frontend/js_src/lib/components/FormMeta/index.tsx#L58

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

**Comment by maxpatiiuk** [comment]
> Not sure what you mean, it is just not returned from useResourceView().
How would this apply to the subforms in app resources?

I should clarified that I meant look at how that place detects if it's in the form meta and do something similar to detect if you are in the form editor in here rather than URL parsing

see https://github.com/specify/specify7/pull/4891#discussion_r1593340322

File: specifyweb/frontend/js_src/lib/components/Forms/Save.tsx (hunk 1 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Forms/Save.tsx:113]: mhhh okok! Much better

**Review by melton-jason — APPROVED** [review]
Very clean! 

Currently, you can also delete resources through the form editor. 
Is this also something we want to disable?

**Comment by CarolineDenis** [comment]
> Very clean!
> 
> Currently, you can also delete resources through the form editor. Is this also something we want to disable?

@grantfitzsimmons ?

**Comment by grantfitzsimmons** [comment]
> > Currently, you can also delete resources through the form editor. Is this also something we want to disable?

@CarolineDenis I think so!

**Review by maxpatiiuk — APPROVED** [review]

**Review by melton-jason — APPROVED** [review]

**Comment by lexiclevenger** [comment]
Previewing record formatters using existing records also allows the user to edit and save those records, as well as create and delete records using the query combo boxes. Is that also something that should be addressed in this PR?

**Comment by CarolineDenis** [comment]
@lexiclevenger, 
We can create a new Issue for that

**Review by combs-a — APPROVED** [review]
- [X] verify save button is disabled 
- [X] verify there is no delete button 

image

image

This looks good, no delete button and save is disabled even when you go through multiple records it seems. Good work!

**Review by alesan99 — APPROVED** [review]
- Go to app resources -> form definition
- Click on any table with a query combo box in the form (e.g. collection object with the accession qcbx)
- In visual editor click the add button for the query combo box
- Fill out some information in the form
- [x] verify save button is disabled 
- [ ] verify there is no delete button 

Can't save or delete as expected!

**Pull request merged**